src/controllers/lesson/lesson.controller.ts

```diff
@@ -1,149 +1,113 @@
 import { NextFunction, Request, Response } from 'express';
-import * as Joi from 'joi';
-
 import { ResponseStatusCodesEnum } from '../../constants';
-import { ErrorHandler, errors } from '../../errors';
+import { calculationPageCount } from '../../helpers/course';
+import { lessonSortingAttributes } from '../../helpers/lesson';
 import { ILesson, IRequestExtended, IUser } from '../../interfaces';
 import { lessonService } from '../../services';
-import { addQuestionToLessonValidator, lessonFilterParamtresValidator, lessonUpdateDataValidator, lessonValidator } from '../../validators';
-
-const lessonSortingAttributes: Array<keyof ILesson> = ['number', 'label', 'tags', '_id'];
 
 class LessonController {
 
-  async createLesson(req: IRequestExtended, res: Response, next: NextFunction) {
-
-    const lessonValue = req.body;
-    const {_id} = req.user as IUser;
+    async createLesson(req: IRequestExtended, res: Response, next: NextFunction) {
 
-    const lessonValidity = Joi.validate(lessonValue, lessonValidator);
+        const lessonValue = req.body;
+        const {_id} = req.user as IUser;
+        await lessonService.createLesson({...lessonValue, user_id: _id});
 
-    if (lessonValidity.error) {
-      return next(new ErrorHandler(ResponseStatusCodesEnum.BAD_REQUEST, lessonValidity.error.details[0].message));
+        res.status(ResponseStatusCodesEnum.CREATED).end();
     }
 
-    await lessonService.createLesson({...lessonValue, user_id: _id});
-
-    res.status(ResponseStatusCodesEnum.CREATED).end();
-  }
-
-  getLessonById(req: IRequestExtended, res: Response, next: NextFunction) {
+    async getLessons(req: Request, res: Response, next: NextFunction) {
 
-    const lesson = req.lesson as ILesson;
+        const {
+            limit = 20,
+            offset = 0,
+            sort = '_id',
+            order,
+            ...filter
+        } = req.query;
 
-    res.json({data: lesson});
-  }
+        lessonSortingAttributes(sort, next);
 
-  async getLesson(req: Request, res: Response, next: NextFunction) {
+        const lesson = await lessonService.getLessons(+limit, +offset, sort, order, filter);
 
-    const {
-      limit = 20,
-      offset = 0,
-      sort = '_id',
-      order,
-      ...filter
-    } = req.query;
+        res.json({
+            data: {
+                lesson,
+                count: lesson.length,
+                pageCount: calculationPageCount(lesson.length, limit)
```

Comment:
Same error with `pageCount`. 
Try to google something `findAndCountAll`
And check al files